File: swift/native/HelloWorldKomencoNative/Sources/Operation.swift
import Foundation

class Operation {
    private var gate: String
    private var qubits: [Int]
    private var params: [Double]

    init(gate: String, qubits: [Int], params: [Double]) {
        self.gate = gate
        self.qubits = qubits
        self.params = params
    }

    func getGate() -> String {
        return gate
    }

    func getQubits() -> [Int] {
        return qubits
    }

    func getParams() -> [Double] {
        return params
    }
}

File: java/native/src/com/automatski/komenco/Operation.java
package com.automatski.komenco;

import java.util.List;

public class Operation {

	private String gate;
	private List<Integer> qubits;
	private List<Double> params;
	
	public Operation(String gate, List<Integer> qubits, List<Double> params) {
		this.gate = gate;
		this.qubits = qubits;
		this.params = params;
		
	}

	public String getGate() {
		return gate;
	}

	public List<Integer> getQubits() {
		return qubits;
	}

	public List<Double> getParams() {
		return params;
	}
	
	
}
